services/termincommittee/term_in_committee.go

```diff
@@ -207,9 +207,8 @@ func (tic *TermInCommittee) initView(ctx context.Context, newView primitives.Vie
 	return current, nil
 }
 
-func (tic *TermInCommittee) Dispose() {
-	tic.electionTrigger.Stop()
-	height := tic.State.Height()
+func (tic *TermInCommittee) Dispose(height primitives.BlockHeight) {
+	tic.electionTrigger.Stop() // smelly - we are not verifying that electionTrigger belongs to the same height
```

Comment:
Agree

Reply:
However, it should not introduce a regression since immediately after `Dispose()` we call `leanhelixterm.NewLeanHelixTerm()` which resets a new trigger for the next term. It looks breakable but should not introduce a new regression bug.  lets discuss